app/functions_for_battle/apply_potion.py

```diff
@@ -0,0 +1,11 @@
+def apply_potion(knight: dict) -> None:
+    if knight["potion"] is not None:
+        if "power" in knight["potion"]["effect"]:
+            knight["power"] += knight["potion"]["effect"]["power"]
+
+        if "protection" in knight["potion"]["effect"]:
+            knight["protection"] += (
+                knight)["potion"]["effect"]["protection"]
```

Comment:
There's a syntax error here. The extra `knight` in parentheses is causing the program to fail. Remove the redundant `knight` to fix the issue.

```suggestion
knight["protection"] += knight["potion"]["effect"]["protection"]
```